### PR TITLE
Enroll PLC users in a specific section

### PR DESCRIPTION
**Pull request by joshlory**
This is a temporary hack to ensure PLC users are enrolled in a specific section for TeacherCon.  In the future we'll generalize this flow to support multiple PLC sections.

### CODE REVIEW

File: dashboard/test/models/section_test.rb [hunk 1]
```diff
@@ -67,4 +67,17 @@ class SectionTest < ActiveSupport::TestCase
     assert !Section.exists?(section.id)
   end
 
+  test "cannot join plc section without a valid enrollment" do
+    user = create(:user)
+
+    @section = create(:section)
```
Comment (joshlory) [inline dashboard/test/models/section_test.rb:73]: Nit: doesn't need to be an instance variable.

File: dashboard/app/models/section.rb [hunk 2]
```diff
@@ -61,6 +61,11 @@ def students_attributes=(params)
   end
 
   def add_student(student)
+    # TODO: generalize to future PLC sections
+    if code && code == Plc::UserCourseEnrollment::PLC_SECTION.try(:code) && !Plc::UserCourseEnrollment.exists?(user: student)
```
Comment (mehalshah) [inline dashboard/app/models/section.rb:65]: Does this need to be a try? If there is no code, that's a problem because it means there is no section PRJWJN

**Comment by mehalshah** [comment]
One question otherwise lgtm

File: dashboard/app/models/section.rb (hunk 2 above)
Comment (joshlory) [inline dashboard/app/models/section.rb:65]: That section only exists in certain environments.

**Pull request merged**